main/file.c

```diff
@@ -382,83 +331,43 @@ static void sub_go_to_sleep(uint64_t interval_ms){
             esp_deep_sleep_start();
         }
     }
-
-}
-
-static bool check_transmit_period(uint64_t interval_ms){
-
-    bool flag = false;
-
-    for(int i = 0; i < ENUM_MAX; i++){
-            
-        flag |= (sub_struct.transmit_periods[i] == interval_ms);
-    }
-
-    return flag;
 }
 
-static uint64_t get_interval_ms(void){
+static inline uint64_t get_interval_ms(){
 
-    uint64_t temp_interval_ms;
-
-    if(xSemaphoreTake(sub_struct.interval_mutex, portMAX_DELAY) == pdTRUE){
-
-        temp_interval_ms = sub_struct.interval_ms;
-
-        xSemaphoreGive(sub_struct.interval_mutex);
-    }
-
-    return temp_interval_ms;
+    return sub_struct.transmit_periods[sub_struct.status];
 }
 
-static void run_transmit_timer(void){
-
-    uint64_t temp_interval_ms = get_interval_ms();
-    TickType_t interval_ticks = pdMS_TO_TICKS(temp_interval_ms);
+static inline bool is_sleep_interval(uint64_t interval_ms){
 
-    TickType_t current_timer_period_ticks = xTimerGetPeriod(sub_struct.transmit_timer);
-
-    if(interval_ticks != current_timer_period_ticks && check_transmit_period(temp_interval_ms)){
-
-        xTimerChangePeriod(sub_struct.transmit_timer, pdMS_TO_TICKS(temp_interval_ms), 0);
-    }
-    else {
-
-        xTimerStart(sub_struct.transmit_timer, 0);
-    }   
+    return interval_ms == sub_struct.transmit_periods[SLEEP_MS];
 }
 
-static bool get_transmit_state(void){
+bool get_transmit_state(){
```

Comment:
Tutaj też można by było coś z nazwą pomyśleć, bo transmit_state trochę mi podsuwa, że ma mi powiedzieć o tym jaki jest status transmisji (czyli tego co wysłaliśmy), a nie czy to już czas na wysłanie paczki czy nie 📦